src/components/BetaModal.js

```diff
@@ -1,56 +1,101 @@
 import PropTypes from 'prop-types'
 import React from 'react'
 
-const BetaModal = ({ onConfirm }) => {
-  const fixed = 'fixed top-0 bottom-0 left-0 right-0'
+import { showFeedback } from '../actions/feedback'
+import { hideModal } from '../actions/modal'
 
-  return (
-    <div className={`${fixed} flex flex-column items-center justify-center z2`}>
-      <div className={`${fixed} bg-black-translucent`} />
+
+class BetaModal extends React.Component {
+  componentDidMount() {
+    this.triggerElement = document.activeElement || document.body
+    document.addEventListener('keydown', this.closeOnEsc)
+    this.setFocus()
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.closeOnEsc)
+  }
+
+  setFocus = () => {
```

Comment:
Ya, when the component is displayed it needs to set focus on the first textarea within itself. It then needs to "trap" the focus meaning that you cannot tab out of the tool when it is expanded. And when it closes, it needs to restore focus to the element that triggered it.